### PR TITLE
Add QFieldGuide component.

### PR DESCRIPTION
**Pull request by mohsenD98**
### Just a quick glance at the made component:

[Screencast from 2024-07-16 18-08-02.webm](https://github.com/user-attachments/assets/51185768-96c9-4120-ae0d-1620765dac11)

_this feature will be added later._

### CODE REVIEW

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 0997941e422f4e176082b5a9423d77491bea5f8f: [**arm64-android**](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield-dev-5441-8a4365860294ae4325037abdd206365d2e4a1c71-android-arm64.apk)
<details>
<summary>Other architectures</summary>

- [arm-neon-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield-dev-5441-8a4365860294ae4325037abdd206365d2e4a1c71-android-arm-neon.apk)
- [x64-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield-dev-5441-8a4365860294ae4325037abdd206365d2e4a1c71-android-x64.apk)
- [x86-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield-dev-5441-8a4365860294ae4325037abdd206365d2e4a1c71-android-x86.apk)
- [**all access arm64-android**](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield_all_access-dev-5441-8a4365860294ae4325037abdd206365d2e4a1c71-android-arm64.apk)
- [all access arm-neon-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield_all_access-dev-5441-8a4365860294ae4325037abdd206365d2e4a1c71-android-arm-neon.apk)
- [all access x64-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield_all_access-dev-5441-8a4365860294ae4325037abdd206365d2e4a1c71-android-x64.apk)
- [all access x86-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield_all_access-dev-5441-8a4365860294ae4325037abdd206365d2e4a1c71-android-x86.apk)
</details>

File: src/qml/QFieldGuide.qml [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+
+  property bool enablePannelAnimation: false
+  property var steps: []
+  property int targetMargins: 5
+  property Component nextButton: nextButton
+  property Component prevButton: prevButton
+  property int index: 0
+  property string finishText: qsTr("Finish")
+  property string nextText: qsTr("Next")
+  property string previousText: qsTr("Previous")
```
Comment (nirvn) [inline src/qml/QFieldGuide.qml:39]: Can we move the property to sit on top of the Popup item?

File: src/qml/QFieldGuide.qml [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+
+  Component {
+    id: nextButton
+    QfButton {
+      text: isEnd ? guide.finishText : guide.nextText
```
Comment (nirvn) [inline src/qml/QFieldGuide.qml:44]: Could we rename this to isLast / isLastItem / isLastText?

File: src/qml/QFieldGuide.qml [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+          guide.close();
+        } else {
+          guide.index = guide.index + 1;
+        }
+      }
+    }
+  }
+
+  Component {
+    id: prevButton
```
Comment (nirvn) [inline src/qml/QFieldGuide.qml:62]: I'd avoid abbreviations here, go for previousButton.

File: src/qml/QFieldGuide.qml [hunk 4]
```diff
@@ -0,0 +1,327 @@
...
+    }
+    Behavior on y  {
+      enabled: enablePannelAnimation
+
+      NumberAnimation {
+        duration: 150
+      }
+    }
+
+    property color color: Theme.mainBackgroundColor
```
Comment (nirvn) [inline src/qml/QFieldGuide.qml:179]: Again, let's make sure properties are sitting at the beginning of items (below ID).

File: src/qml/QFieldGuide.qml [hunk 5]
```diff
@@ -0,0 +1,327 @@
...
+      anchors.top: description.bottom
+      anchors.left: parent.left
+      fillMode: AnimatedImage.PreserveAspectFit
+      anchors.leftMargin: 14
+      anchors.topMargin: 8
+    }
+
+    Loader {
+      id: nextButtonLoader
+      property bool isEnd: guide.index === steps.length - 1
```
Comment (nirvn) [inline src/qml/QFieldGuide.qml:257]: See isLast comment above.

File: src/qml/QFieldGuide.qml [hunk 6]
```diff
@@ -0,0 +1,327 @@
...
+      sourceComponent: nextButton
+      anchors {
+        bottom: parent.bottom
+        bottomMargin: 8
+        right: parent.right
+        rightMargin: 15
+      }
+    }
+    Loader {
+      id: prevButtonLoader
```
Comment (nirvn) [inline src/qml/QFieldGuide.qml:267]: previousButtonLoader

File: src/qml/qgismobileapp.qml [hunk 7]
```diff
@@ -38,6 +38,40 @@ ApplicationWindow {
...
+    id: tour
+    steps: [{
+        "title": qsTr("Zoom In Button"),
+        "description": qsTr("Main responsibility is described."),
+        "target": () => zoomInButton
+      }, {
+        "title": qsTr("Zoom Out Button"),
+        "description": qsTr("Main responsibility is described."),
+        "target": () => zoomOutButton
+      }, {
```
Comment (nirvn) [inline src/qml/qgismobileapp.qml:60]: As discussed in the chat, let's do one step for both zoom in and out buttons instead.

**Review by nirvn — COMMENTED** [review]
Very, very good work. Your knowledge of QML is shining through.

**Comment by mohsenD98** [comment]
> Very, very good work. Your knowledge of QML is shining through.

your words means a lot to me, thank you ;)

File: src/qml/QFieldGuide.qml (hunk 1 above)
Comment (mohsenD98) [reply src/qml/QFieldGuide.qml]: yes, done.

File: src/qml/QFieldGuide.qml (hunk 2 above)
Comment (mohsenD98) [reply src/qml/QFieldGuide.qml]: done :+1:

File: src/qml/QFieldGuide.qml (hunk 3 above)
Comment (mohsenD98) [reply src/qml/QFieldGuide.qml]: done :+1:

File: src/qml/QFieldGuide.qml (hunk 5 above)
Comment (mohsenD98) [reply src/qml/QFieldGuide.qml]: done :+1:

File: src/qml/QFieldGuide.qml (hunk 6 above)
Comment (mohsenD98) [reply src/qml/QFieldGuide.qml]: done :+1:

File: src/qml/qgismobileapp.qml (hunk 7 above)
Comment (mohsenD98) [reply src/qml/qgismobileapp.qml]: completely removed

**Review by nirvn — APPROVED** [review]

**Comment by suricactus** [comment]
Kudos for this feature!

**Pull request merged**